src/golang/cmd/migrator/versions/000016_add_artifact_type_column_to_artifact/up_postgres.go

```diff
@@ -1,9 +1,11 @@
 package _000016_add_artifact_type_column_to_artifact
 
-const upPostgresScript = `
+const upPostgresAddColumn = `
```

Comment:
When we add in lazy execution, there may be cases that these type fields are populated after the first workflow run, so I think it makes sense to keep it flexible here.

Reply:
IF that's the case then do we need to change the `artifact.type` field to be `Nullable`?

Reply:
Now that I think about it, in lazy mode we will want to populate the initial type value of an artifact to be "untyped", so in that sense they are not null. But if we add a not null constraint here it will complain because existing rows in the database has an empty "type" field. I Googled a bit and there doesn't seem to be a way to add not null constraint later in sqlite other than recreating the table..

Reply:
Let me try if default value works. I'm trying to set the initial value to "untyped"

Reply:
Yep not null with default worked.